plugin-server/src/utils/db/db.ts

```diff
@@ -465,6 +476,78 @@ export class DB {
+        // Query from postgres and update cache
+        const result = await this.postgresQuery(
+            'SELECT person_id FROM posthog_persondistinctid WHERE team_id=$1 AND distinct_id=$2 LIMIT 1',
+            [teamId, distinctId],
+            'fetchPersonId'
+        )
+        if (result.rows.length !== 0) {
+            const personId = result.rows[0].person_id as number
+            await this.updatePersonIdCache(teamId, distinctId, personId)
+            return personId
+        }
+        return null
+    }
+
+    public async getPersonInfoThroughCache(
+        teamId: number,
+        personId: number
+    ): Promise<[string | null, DateTime | null, Properties | null]> {
+        if (!this.personInfoCachingEnabledTeams.has(teamId)) {
+            return [null, null, null]
+        }
+        const [personUuid, personCreatedAt, personProperties] = await Promise.all([
+            this.redisGet(this.getPersonUuidCacheKey(teamId, personId), null),
+            this.redisGet(this.getPersonCreatedAtCacheKey(teamId, personId), null),
+            this.redisGet(this.getPersonPropertiesCacheKey(teamId, personId), null),
+        ])
+        if (personUuid !== null && personCreatedAt !== null && personProperties !== null) {
+            this.statsd?.increment(`person_info_cache.hit`, { lookup: 'person_properties', team_id: teamId.toString() })
+            return [personUuid as string, personCreatedAt as DateTime, personProperties as Properties]
+        }
+        this.statsd?.increment(`person_info_cache.miss`, { lookup: 'person_properties', team_id: teamId.toString() })
+        // Query from postgres and update cache
+        const result = await this.postgresQuery(
+            'SELECT uuid, created_at, properties FROM posthog_person WHERE team_id=$1 AND id=$2 LIMIT 1',
+            [teamId, personId],
+            'fetchPersonProperties'
+        )
+        if (result.rows.length !== 0) {
+            const personUuid = result.rows[0].uuid as string
+            const personCreatedAt = result.rows[0].created_at as DateTime
+            const personProperties = result.rows[0].properties as Properties
+            await Promise.all([
+                this.updatePersonUuidCache(teamId, personId, personUuid),
+                this.updatePersonCreatedAtCache(teamId, personId, personCreatedAt),
+                this.updatePersonPropertiesCache(teamId, personId, personProperties),
+            ])
+            return [personUuid, personCreatedAt, personProperties]
+        }
+        return [null, null, null]
+    }
+
+    public async getPersonInfoThroughCacheFromDistinctId(
+        teamId: number,
+        distinctId: string
+    ): Promise<[string | null, DateTime | null, Properties | null]> {
+        const personId = await this.getPersonIdThroughCache(teamId, distinctId)
+        if (personId !== null) {
+            return await this.getPersonInfoThroughCache(teamId, personId)
+        }
+        return [null, null, null]
```

Comment:
I think I'd rather return an object as that makes it much easier to work with instead of having to remember the order of the returned values